paddle/fluid/operators/detection/collect_fpn_proposals_op.cc

```diff
@@ -0,0 +1,118 @@
+    PADDLE_ENFORCE(context->HasInputs("MultiLayerRois"),
+                   "Inputs(MultiLayerRois) shouldn't be null");
+    PADDLE_ENFORCE(context->HasInputs("MultiLayerScores"),
+                   "Inputs(MultiLayerScores) shouldn't be null");
+    PADDLE_ENFORCE(context->HasOutput("FpnRois"),
+                   "Outputs(MultiFpnRois) of DistributeOp should not be null");
+    auto roi_dims = context->GetInputsDim("MultiLayerRois");
+    auto score_dims = context->GetInputsDim("MultiLayerScores");
+    auto post_nms_topN = context->Attrs().Get<int>("post_nms_topN");
+    std::vector<int64_t> out_dims;
+    for (auto &roi_dim : roi_dims) {
+      PADDLE_ENFORCE_EQ(roi_dim[1], 4,
+                        "Second dimension of Input(MultiLayerRois) must be 4");
+    }
+    for (auto &score_dim : score_dims) {
+      PADDLE_ENFORCE_EQ(
+          score_dim[1], 1,
+          "Second dimension of Input(MultiLayerScores) must be 1");
+    }
+    context->SetOutputDim("FpnRois", {post_nms_topN, 4});
+    if (!context->IsRuntime()) {  // Runtime LoD infershape will be computed
+      // in Kernel.
+      context->ShareLoD("MultiLayerRois", "FpnRois");
+    }
+    if (context->IsRuntime()) {
+      std::vector<framework::InferShapeVarPtr> roi_inputs =
+          context->GetInputVarPtrs("MultiLayerRois");
+      std::vector<framework::InferShapeVarPtr> score_inputs =
+          context->GetInputVarPtrs("MultiLayerScores");
+      framework::LoD lod;
+      for (size_t i = 0; i < roi_inputs.size(); ++i) {
+        framework::Variable *roi_var =
+            boost::get<framework::Variable *>(roi_inputs[i]);
+        framework::Variable *score_var =
+            boost::get<framework::Variable *>(score_inputs[i]);
+        auto &roi_lod = roi_var->Get<LoDTensor>().lod();
+        auto &score_lod = score_var->Get<LoDTensor>().lod();
+        PADDLE_ENFORCE_EQ(roi_lod, score_lod,
+                          "Inputs(MultiLayerRois) and Inputs(MultiLayerScores) "
+                          "should have same lod.");
+        if (lod.size() == 0) {
+          lod = roi_lod;
+        }
+        PADDLE_ENFORCE_EQ(
+            lod.size(), roi_lod.size(),
+            "Each of Inputs(MultiLayerRois) should have same lod size.");
+        PADDLE_ENFORCE_EQ(
+            lod.size(), score_lod.size(),
+            "Each of Inputs(MultiLayerScores) should have same lod size.");
+      }
+    }
+  }
+
+ protected:
+  framework::OpKernelType GetExpectedKernelType(
+      const framework::ExecutionContext &ctx) const override {
+    auto data_type =
+        framework::GetDataTypeOfVar(ctx.MultiInputVar("MultiLayerRois")[0]);
+    return framework::OpKernelType(data_type, ctx.GetPlace());
+  }
+};
+
+class CollectFpnProposalsOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("MultiLayerRois",
+             "(LoDTensor) Multiple roi LoDTensors from each levels in shape "
```

Comment:
Each item is a LoDTensor from each level ...

roi -> RoI 
each levels -> each level

Reply:
Done